### PR TITLE
#437 Added back button for all admin/dashboard pages

### LINKED ISSUE #437: [ADMIN PANEL] Go back button
For easier navigation within the admin panel

### PR DESCRIPTION
**Pull request by Pastedko**
#437 Added back button for all admin/dashboard/* pages(#437 )

Created a back button component in back_button.jsx and a new file with routes that have the back button. Also had to edit the css for the webshortener page and s3 page

resolves #437 

## How to verify:

- [ ] Start the React frontend, Golang backend, Python backend

- [ ] Open your browser or API client and head to /admin/dashboard.

- [ ] Open any page

- [ ] Click the back button in the top-left corner

- [ ] Button also exists for the subpages (add and action pages), so check it out

### CODE REVIEW

File: packages/web/src/components/admin_page/url_shortener/table.jsx [hunk 1]
```diff
@@ -138,7 +138,7 @@ const UrlsTable = () => {
                     Create a shortened URL
                 </Button>
             </OverlayTrigger>
-            <Table style={{ color: 'white' }}>
+            <Table style={{ color: 'white', marginTop: '4vw' }}>
```
Comment (asynchroza) [inline packages/web/src/components/admin_page/url_shortener/table.jsx:141]: Why is this needed?

File: packages/web/src/components/admin_page/s3_page/s3.css [hunk 2]
```diff
@@ -5,7 +5,8 @@
 
 .s3-upload-image {
     background-color: gray;
-    height: 80vh;
+    padding-top: 4vw;
```
Comment (asynchroza) [inline packages/web/src/components/admin_page/s3_page/s3.css:8]: Why is this needed?

File: packages/web/src/App.js [hunk 3]
```diff
@@ -64,47 +34,12 @@ function App() {
...
                 <Route path="/hackaubg" element={<HackAUBG />} />
                 {featureSwitches.jobs ? <Route path="/jobs" element={<JobsSection />} /> : null}
-                <Route path="/*" element={<NotFound />} />
-                <Route path="/admin/dashboard/s3" element={<S3Panel />} />
-                <Route path="/admin/dashboard/s3/objects" element={<RenderStorageObjects />} />
-                <Route path="/admin/dashboard/shortener" element={<UrlsTable />} />
-                <Route path="/admin/dashboard/fswitches" element={<RenderSwitches />} />
+                <Route path="/admin" element={<LandingAdminPage />} />
+                <Route path="/admin/dashboard" element={<Dashboard />} />
+                <Route path="/admin/dashboard/*" element={<BackBtnAdmin />} />
```
Comment (asynchroza) [inline packages/web/src/App.js:42]: That's an interesting approach. I like it.

File: packages/web/src/components/admin_page/url_shortener/table.jsx (hunk 1 above)
Comment (Pastedko) [reply packages/web/src/components/admin_page/url_shortener/table.jsx:141]: Otherwise the back button will appear on top of the text of the page. With the padding it moves it down a bit so that the button does not overlap with anything

File: packages/web/src/Admin_dashboard_routes.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        <Route path="jobs" element={<BackBtnAdd Component={RenderJobs} />} />
+        <Route path="jobs/actions" element={<BackBtnAdd Component={JobActions} />} />
+        <Route path="jobs/add" element={<BackBtnAdd Component={AddJobs} />} />
+        <Route path="s3" element={<BackBtnAdd Component={S3Panel} />} />
+        <Route path="s3/objects" element={<BackBtnAdd Component={RenderStorageObjects} />} />
+        <Route path="shortener" element={<BackBtnAdd Component={UrlsTable} />} />
+        <Route path="fswitches" element={<BackBtnAdd Component={RenderSwitches} />} />
+    </Routes>
+);
+const BackBtnAdd = ({ Component }) => {
```
Comment (asynchroza) [inline packages/web/src/Admin_dashboard_routes.js:75]: Rename to `AdminContainerWithButton`

File: packages/web/src/components/admin_page/s3_page/s3.css (hunk 2 above)
Comment (Pastedko) [reply packages/web/src/components/admin_page/s3_page/s3.css:8]: Otherwise the back button will appear on top of the text of the page. With the padding it moves it down a bit so that the button does not overlap with anything

File: packages/web/src/Admin_dashboard_routes.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    return (
+        <div>
```
Comment (asynchroza) [inline packages/web/src/Admin_dashboard_routes.js:77]: Instead of a div, use either `React.Fragment` or `<></>`.

File: packages/web/src/Admin_dashboard_routes.js (hunk 4 above)
Comment (asynchroza) [inline packages/web/src/Admin_dashboard_routes.js:75]: Only component names are capitalized. Rename the prop to `component` or `element`.

File: packages/web/src/Admin_dashboard_routes.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+import AddTeamMember from './components/admin_page/hackathon/hackathon_team_members/new_member';
+import AddNewTeam from './components/admin_page/hackathon/hackathon_teams/new_team.jsx';
+import RenderNoTeamParticipants from './components/admin_page/hackathon/hackathon_no_team_participants/render_no_team_participants';
+import NoTeamParticipantsActions from './components/admin_page/hackathon/hackathon_no_team_participants/single_no_team_participant';
+import AddNoTeamParticipant from './components/admin_page/hackathon/hackathon_no_team_participants/new_no_team_participant';
+import UrlsTable from './components/admin_page/url_shortener/table';
+import { Routes, Route } from 'react-router-dom';
+import BackBtn from './components/admin_page/back_button';
+
+const BackBtnAdmin = () => (
```
Comment (asynchroza) [inline packages/web/src/Admin_dashboard_routes.js:35]: Rename to `AdminRoutesWithBackButton`

File: packages/web/src/Admin_dashboard_routes.js [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+    <Routes>
+        <Route path="members" element={<BackBtnAdd Component={RenderMembers} />} />
```
Comment (asynchroza) [inline packages/web/src/Admin_dashboard_routes.js:37]: I suggest you do the following here:

```javascript

const adminRoutes = [
    {
        path: 'members',
        element: <RenderMembers />,
    },
    {
        path: 'members/actions',
        element: <MemberActions />,
    },
   // add other routes here
];

const AdminRoutesWithBackButton = () => (
    <Routes>
        {adminRoutes.map(route => (
            <Route
                path={route.path}
                element={<AdminContainerWithButton component={route.element} />}
                key={route.path}
            />
        ))}
    </Routes>
);

const AdminContainerWithButton = ({ component }) => {
    return (
        <>
            <BackBtn />
            {component}
        </>
    );
};
export default AdminRoutesWithBackButton;

```

File: packages/web/src/Admin_dashboard_routes.js [hunk 8]
```diff
@@ -0,0 +1,83 @@
+import MemberActions from './components/admin_page/members_page/single_member';
```
Comment (asynchroza) [inline packages/web/src/Admin_dashboard_routes.js:1]: Rename file to `AdminRoutesWithBackButton.js`.

Let's stick to naming the files using PascalCase. That's the usual convention when it comes to files with jsx in them. You may use `snake_case` for json files. 


@Aleks1908 you should enforce this convention

File: packages/web/src/components/admin_page/url_shortener/table.jsx (hunk 1 above)
Comment (asynchroza) [reply packages/web/src/components/admin_page/url_shortener/table.jsx:141]: Isn't there a way to fix this within the button's styling?

Overlapping button is still present. 

image
image

**Comment by asynchroza** [comment]
A couple of nits and this PR should be good to go. Overall good job. Really impressed with your work @Pastedko. You're getting the hang of React rather quickly. Keep it up!

File: packages/web/src/Admin_dashboard_routes.js (hunk 4 above)
Comment (Pastedko) [reply packages/web/src/Admin_dashboard_routes.js:75]: But if is not capitalized it does not accept is as a React component.
image
image
Do you know of a way around it? I tried searching but did not find anything

Comment (asynchroza) [reply packages/web/src/Admin_dashboard_routes.js:75]:
```javascript
const AdminContainerWithButton = ({ component }) => {
    return (
        <>
            <BackBtn />
            {component}
        </>
    );
};
```

Comment (asynchroza) [reply packages/web/src/Admin_dashboard_routes.js:75]: Remind me next time to tell you a little bit more about what children are and how to pass them.

File: packages/web/src/components/admin_page/url_shortener/table.jsx (hunk 1 above)
Comment (Pastedko) [reply packages/web/src/components/admin_page/url_shortener/table.jsx:141]: Now that you opened the mobile view. Do we need a back button for the mobile view? It seems a bit unnecessary and impractital.

Comment (asynchroza) [reply packages/web/src/components/admin_page/url_shortener/table.jsx:141]: You have a point. Disable it on mobile resolutions.

File: packages/web/src/AdminRoutesWithBackButton.js [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+                element={<AdminContainerWithButton component={route.element} path={route.path} />}
+                key={route.path}
+            />
+        ))}
+    </Routes>
+);
+const AdminContainerWithButton = ({ component, path }) => {
+    var position = 'middle';
+    console.log(path);
+    if (path == 'shortener') position = 'top';
```
Comment (asynchroza) [inline packages/web/src/AdminRoutesWithBackButton.js:175]: Avoid using var. Use `const` instead and a ternary operator.

```javascript
const position = path === 'shortener' ? 'top' : 'middle'
```

File: packages/web/src/AdminRoutesWithBackButton.js [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+    },
+    {
+        path: 'shortener',
+        element: <UrlsTable />,
+    },
+    {
+        path: 'fswitches',
+        element: <RenderSwitches />,
+    },
+];
```
Comment (asynchroza) [inline packages/web/src/AdminRoutesWithBackButton.js:160]: You should space out your code. I've seen you do this quite a bit. Make sure to insert at least one newline between code blocks.

File: packages/web/src/AdminRoutesWithBackButton.js (hunk 9 above)
Comment (asynchroza) [reply packages/web/src/AdminRoutesWithBackButton.js:175]: Let's add a prop called `positionButtonOnTop` to the button. This way the button can be reused in other components which can't provide a path.

```javascript
<BackBtn positionButtonOnTop={path === 'shortener'}/>

...

const className = positionButtonOnTop ? "top" : "middle";
...
<Button className={className}/>
```

File: packages/web/src/AdminRoutesWithBackButton.js [hunk 11]
```diff
@@ -169,15 +170,14 @@ const AdminRoutesWithBackButton = () => (
...
 );
+
 const AdminContainerWithButton = ({ component, path }) => {
-    var position = 'middle';
-    console.log(path);
-    if (path == 'shortener') position = 'top';
     return (
         <>
-            <BackBtn style={position} />
+            <BackBtn positionButtonOnTop={path == 'shortener'} />
```
Comment (asynchroza) [inline packages/web/src/AdminRoutesWithBackButton.js:177]: Use the strict equality operator

File: packages/web/src/components/admin_page/back_button.jsx [hunk 12]
```diff
@@ -1,12 +1,15 @@
 import { useNavigate } from 'react-router-dom';
 import { Button } from 'react-bootstrap';
 import './back_button.css';
-const BackBtn = ({ style }) => {
+
+const BackBtn = ({ positionButtonOnTop }) => {
```
Comment (asynchroza) [inline packages/web/src/components/admin_page/back_button.jsx:5]: Consider giving the prop a default value of `false`
